### ISSUE TITLE
Matching set - ALL EXISTS

### ISSUE THREAD

**Issue by zakwillis** [Seddryck/NBi#294]
Hi, there is one feature that seems to be lacking and that is where we have set A being compared to set B and that set A resolves to set B but doesn't fail if there are additional set B rows. 

Simple example:-
```
Set A
1
2
```
```
Set B
1
2
3
```
In this example Set A would pass because the data is in the other set. Am I missing something?

**Comment by FuegoArtificial** [comment]
Hi,

which type of entities are you comparing? Rows in a result set or objects of a cube or...?
Depending on what you want to compare, there might be a different syntax. I believe for each one, you can have an exact comparison as well with an "equal-to" assertion. Therefore your sample test would fail as you expect.
Did I get your question right? Did you check the documentation already at http://www.nbi.io/docs/home/?

Cheers

**Comment by zakwillis** [comment]
Hi there, thanks for the fast reply. I am running this for SQL and I can write a pseudo example.

I would expect this to work because all rows in set a exists in set b.
```sql
select field
from
(select 
1 field
union all
select 
2 field
) a
```
```sql
select field
from
(select 
1 field
union all
select 
2 field
union all
select 
3 field
) b
```

Many thanks,

Zak

**Comment by FuegoArtificial** [comment]
Two ideas:
1, I don't know if the predicate "subsetOf" instead if "equalTo" works, but you could try?
2. you could write a query which takes the first query result and applies an except of the second query result. For the assert you could take any of the both queries with a WHERE 1=0 to get an empty result set. If the first query is a subset of the second query, you will get an empty result set in the system under test-part, too, and the test will pass.

**Comment by Seddryck** [comment]
The constraints ```subsetOf``` (and ```contains```) are not implemented for the comparison of two queries. But this is clearly in the radar and we already have two other requests for these features: #196 and #274. I'll clearly work on this soon.

Closing the issue as duplicate, feel free to add any additional comments to issue #196 or re-open it if your case is different.

**Comment by zakwillis** [comment]
Many thanks Seddryck. I will keep a keen eye on this. When do you think the release will be?

**Comment by Seddryck** [comment]
It's available in a preview of release 1.16: https://ci.appveyor.com/project/Seddryck/nbi/build/1.16.0-ci.1
